GameServerLib/GameObjects/AttackableUnits/AttackableUnit.cs

```diff
@@ -28,6 +28,8 @@ public class AttackableUnit : GameObject, IAttackableUnit
         public int KillDeathCounter { get; set; }
         public int MinionCounter { get; protected set; }
         public IReplication Replication { get; protected set; }
+        public float PhyShieldAmount { get; set; }
```

Comment:
Use an array: `ShieldAmount[]` enumerated by the enum (e.g. `ShieldAmount[(int) SHIELD_PHYSICAL]`)

Reply:
you mean like this?
```        private ShieldType[] ShieldAmount = new ShieldType[2] { ShieldType.SHIELD_MAGICAL, ShieldType.SHIELD_PHYSICAL };```
doesn't know array much, so i don't know the exact setting of it.

Reply:
and will be like this?
```if (ShieldAmount[1] > 0 && type == DamageType.DAMAGE_TYPE_MAGICAL)
            {
                float shieldAmount = ShieldAmount[1];
                ApplyShield(-damage, ShieldType.SHIELD_MAGICAL, false);
                damage -= shieldAmount;
                if (ShieldAmount[1] < 0)
                {
                    ShieldAmount[1] = 0;
                }
            }
            if (ShieldAmount[0] > 0 && type == DamageType.DAMAGE_TYPE_PHYSICAL)
            {
                float shieldAmount = ShieldAmount[0];
                ApplyShield(-damage, ShieldType.SHIELD_PHYSICAL, false);
                damage -= shieldAmount;
                if (ShieldAmount[0] < 0)
                {
                    ShieldAmount[0] = 0;
                }
            }
```
or there's other way can do?

Reply:
Not like this. 
The initialization process should be like this:
`private int[] ShieldAmount = new int[Enum.GetNames(typeof(ShieldType)).Length];`

and then:
 ```if (ShieldAmount[1] > 0 && type == DamageType.DAMAGE_TYPE_MAGICAL)
{
      shieldType = ShieldType.SHIELD_MAGICAL
}
else if ...
{
     shieldType = ShieldType.SHIELD_PHYSICAL
}
```

and only then the remaining code:
```
if (ShieldType != null)
{
        float shieldAmount = ShieldAmount[shieldType];
                ApplyShield(-damage, shieldType, false);
                damage -= shieldAmount;
                if (ShieldAmount[shieldType]< 0)
                {
                    ShieldAmount[shieldType]= 0;
                }
}
```

This is of course just a sketch and you should complete the rest.